test/e2e/framework/pod/exec_util.go

```diff
@@ -198,7 +198,9 @@ func execute(ctx context.Context, url *url.URL, config *restclient.Config, stdin
 			framework.Logf("fallback to secondary dialer from primary dialer err: %v", err)
 			return true
 		}
-		framework.Logf("unexpected error trying to use websockets for pod exec: %v", err)
+		if err != nil {
```

Comment:
@seans3 should this go in the fallback executor instead:

```diff
-	if f.shouldFallback(err) {
+ 	if err != nil && f.shouldFallback(err) {
```

Reply:
I had started wondering about this, too. It's odd that `shouldFallback` gets called when nothing bad happened. @aojea's expectation that the method only gets called when there is a non-nil error is reasonable.

Reply:
Done. I've updated the commit and PR description accordingly.